Froger/MainLoop.java

```diff
@@ -0,0 +1,121 @@
+package Froger;
+
+import javafx.animation.Animation;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.ImagePattern;
+import javafx.scene.text.Font;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+
+public class MainLoop extends Application {
```

Comment:
pamietaj o zasadach pisania czystego kodu w szczegolnosci o ``SOLID``. Tutaj MainLoop robi zdecydowanie za wiele, oprocz tego ze jest glowna petla w kodzie. Wystarczy ogarnac importy, Frog, Cars, Gold to raczej klasy ktore powinny byc raczej w klasie GameEngine albo FrogGame tak zeby MainLoop najlepiej wywolywal tylko ``start()``